modalsheet/src/main/java/eu/wewox/modalsheet/ModalSheet.kt

```diff
@@ -169,12 +170,44 @@ private fun ModalSheet(
             sheetBackgroundColor = backgroundColor,
             sheetContentColor = contentColor,
             scrimColor = scrimColor,
-            sheetContent = content,
+            sheetContent = {
+                ModalSheetScopeImpl(sheetState).content()
+            },
             content = { /* Empty */ }
         )
     }
 }
 
+/**
+ * The scope for the modal sheet content.
+ */
+@ExperimentalSheetApi
+public interface ModalSheetScope {
+
+    /**
+     * Normalized fraction from 0.0 to 1.0, where 0.0 is fully collapsed and 1.0 is fully expanded.
+     */
+    public val normalizedFraction: Float
+        @FloatRange(from = 0.0, to = 1.0)
+        get() = with(swipeState) {
+            when {
+                progress.from == ModalBottomSheetValue.Expanded && progress.to == ModalBottomSheetValue.Expanded ->
+                    1f
+                progress.from == ModalBottomSheetValue.Hidden && progress.to == ModalBottomSheetValue.Hidden ->
+                    1f
+                direction <= 0 ->
+                    progress.fraction
+                else ->
+                    1f - progress.fraction
+            }
+        }
+
+    /**
+     * The swipeable state to read values from.
+     */
+    public val swipeState: SwipeableState<ModalBottomSheetValue>
```

Comment:
But how would you access this state from outside of the bottom sheet? You would have to sync it to external state. Is hoisted state not a suitable option?

Reply:
Now user have an option to use `ModalSheet` with `sheetState` overload, so they can observe the progress even outside of the bottom sheet. However for other overloads I think it is fine to provide it at least as a scope of the content.

Reply:
Yeah, I guess that it would be weird to have two states, one Material and some other one. But it still seems weird to have these things as a context properties. It could as well be extension functions to `ModalBottomSheetState`.
I guess what I am saying is that it does not seem to bring as much value to be inside this library for me.

Reply:
So you would not add any observability at all? So that if users want to observe modal sheet state they must use the `sheetState` overload and for other two it is not supported?

Reply:
I don't know what is the best solution. Just that this seems kinda awkward to me :/ 
Which does not mean that we should not merge this.

Reply:
Yea, but anyway, we can close this for know.

If anyone want to use it, they may use `sheetState` 🤷